tastyworks/tastyworks_api/tw_api.py

```diff
@@ -0,0 +1,647 @@
+import aiohttp
+import inspect
+
+from os import environ
+from datetime import date
+from urllib.parse import quote
+from typing import Dict
+
+
+async def is_error(resp, valid_list: list = None, error_list: list = None):
+    # TODO: Improve to capture errors? Can use the 'ok' data of resp
+    if valid_list is None:
+        valid_list = [200, 201, 202, 204]
+    if error_list is None:
+        error_list = []
+
+    if resp.status not in valid_list or resp.status in error_list:
+        print(
+            f'###################################################\n'
+            f'API returned an error for: {resp.request_info.url}\n'
+            f'Called from: {inspect.stack()[2].function}\n'
+            f'Response status: {resp.status}\n'
+            f'Reason: {resp.reason}\n'
+            f'###################################################'
+        )
+        return False
+    else:
+        return True
+
+
+def make_api_url(key: str, **kwargs):
+    """
+    This will format all the API strings based on the **kwargs and pick the one to use based on the 'key'
+
+    Args:
+        key (str): The key to choose and return from the dictionary
+        **kwargs: Input data needed depending on the type of request
+
+    Returns:
+        A formatted URL string made for the requested API call
+    """
+    symbol = kwargs.get('symbol')
+    account = kwargs.get('account')
+    start_date = kwargs.get('start_date')
+    end_date = kwargs.get('end_date')
+    tag = kwargs.get('tag')
+    order_id = kwargs.get('order_id')
+    entry_id = kwargs.get('entry_id')
+    alert_id = kwargs.get('alert_id')
+    journal_page_offset = kwargs.get('journal_page_offset')
+    watchlist_name = kwargs.get('watchlist_name')
+    per_page = kwargs.get('per_page')
+    page_number = kwargs.get('page_number')
+
+    if not account:
+        account = environ.get('TW_ACCOUNT', '')
+
+    if type(symbol) is list:
+        symbol = ','.join(symbol)
+
+    if per_page:
+        if key == 'TRANSACTIONS' and per_page > 2000:  # Transactions are limited to 2000
+            print('Limited to 2000 items per page for transactions. Setting at 2000 per page.')
+            per_page = 2000
+        if key == 'ORDERS' and per_page > 200:  # Orders are limited to 200
+            print('Limited to 200 items per page for orders. Setting at 200 per page.')
+            per_page = 200
+
+    root = 'https://api.tastyworks.com'
+
+    crumbs = {
```

Comment:
If you retrieved accounts via the `TradingAccount` model, these would need to be rewritten programmatically since `{account}` would be a list now rather than a single string.  See above about accounts.